src/main.cpp

```diff
@@ -300,15 +300,20 @@ void parseHeader() {
         const auto IFACE_CLASS_NAME_CAMEL = camelize((clientCode ? "CC_" : "C_") + iface.name);
 
         // begin the class
-        HEADER +=
-            std::format(R"#(
+        HEADER += std::format(R"#(
+struct {}DestroyWrapper {{
+    wl_listener listener;
+    {}* parent;
```

Comment:
could you initialize this to `nullptr`? Just in case.

Reply:
changed to
```
{}* parent = nullptr;
```